lib/base/functionwrapper.hpp

```diff
@@ -39,7 +39,7 @@ inline std::function<Value (const std::vector<Value>&)> WrapFunction(void (*func
 template<typename Return>
 std::function<Value (const std::vector<Value>&)> WrapFunction(Return (*function)(const std::vector<Value>&))
 {
-	return std::bind(function, _1);
+	return [function](const std::vector<Value>& wrappers) -> Return { return function(wrappers); };
```

Comment:
Are you sure about the return type `Return`?

Reply:
Yes, obviously you can see how the function pointer is declared with its return type.

Reply:
Have a look at the return type of WrapFunction. Still sure?

Reply:
However, the lambda does not return WrapFunction return type but the one from the function pointer.

Reply:
*, you could also take ``Value`` as return type.

Reply:
Yes, _your lambda currently_ returns a `Return`.

Describe WrapFunction's return type with your own words.

Reply:
> *, you could also take `Value` as return type.

Try that.